src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -173,6 +205,57 @@ uint_t reorder_qubits(const reg_t qubits, uint_t index) {
   return new_index;
 }
 
+template <class T>
+void permute_all_qubits(const std::vector<T>& orig_statevector, 
+			const reg_t &input_qubits,
+			const reg_t &output_qubits,
+			std::vector<T>& new_statevector) {
```

Comment:
This function, and other functions (some of them are already existing, like `squeeze_qubits`), are not related to the MPS structure, and as such should be located elsewhere (not necessarily in this PR).

Reply:
I agree, but since I developed it for this PR, it is here.